src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -22,7 +21,72 @@ public abstract class JsonPropertyInfo
 
         internal ConverterStrategy ConverterStrategy;
 
-        internal abstract JsonConverter ConverterBase { get; set; }
+        /// <summary>
+        /// Converter resolved from PropertyType and not taking in consideration any custom attributes or custom settings.
+        /// - for reflection we store the original value since we need it in order to construct typed JsonPropertyInfo
+        /// - for source gen it remains null, we will initialize it only if someone used resolver to remove CustomConverter
+        /// </summary>
+        internal JsonConverter? NonCustomConverter { get; set; }
```

Comment:
Isn't this just equivalent to doing `_jsonTypeInfo.Converter`?

Reply:
yes but JsonTypeInfo currently uses `PropertyInfoForTypeInfo.NonCustomConverter` for storing converter (even without this change) and it would be a bit of refactoring to change that so I'm preserving that behavior here